build.rs

```diff
@@ -1,17 +1,127 @@
-extern crate structopt;
+fn main() {
+	//create the shell completions
+	shell_completions::generate();
+
+	//create seccomp.bpf for target_arch
+	seccomp::generate();
+}
 
-use structopt::clap::App;
-use structopt::clap::Shell;
+mod shell_completions {
+	extern crate structopt;
 
-include!("src/cli_args.rs");
+	use structopt::clap::Shell;
 
-fn main() {
-	let directory = match std::env::var_os("COMPLETIONS_DIR") {
-		None => return,
-		Some(out_dir) => out_dir,
-	};
-	let mut app: App = CliArgs::clap();
-	app.gen_completions(env!("CARGO_PKG_NAME"), Shell::Bash, &directory);
-	app.gen_completions(env!("CARGO_PKG_NAME"), Shell::Fish, &directory);
-	app.gen_completions(env!("CARGO_PKG_NAME"), Shell::Zsh, &directory);
+	include!("src/cli_args.rs");
+
+	pub fn generate() {
+		let directory = match std::env::var_os("COMPLETIONS_DIR") {
+			None => return,
+			Some(out_dir) => out_dir,
+		};
+		let mut app = CliArgs::clap();
+		app.gen_completions(env!("CARGO_PKG_NAME"), Shell::Bash, &directory);
+		app.gen_completions(env!("CARGO_PKG_NAME"), Shell::Fish, &directory);
+		app.gen_completions(env!("CARGO_PKG_NAME"), Shell::Zsh, &directory);
+	}
+}
+
+mod seccomp {
+	use libscmp::{resolve_syscall_name, Action, Arch, Filter};
+	use std::{fs::File, path::Path};
+	use std::{os::unix::io::IntoRawFd, str::FromStr};
+
+	pub fn generate() {
+		let mut ctx = Filter::new(Action::Allow).unwrap();
+
+		//Get the target_arch configured in cargo to allow cross compiling
+		let target_arch = std::env::var("CARGO_CFG_TARGET_ARCH").expect(
+			"Compiling without cargo is not supported! Env CARGO_CFG_TARGET_ARCH not found.",
+		);
+		let target_arch = Arch::from_str(&target_arch)
+			.expect("CARGO_CFG_TARGET_ARCH is not a supported seccomp architecture!");
+		ctx.remove_arch(Arch::NATIVE).unwrap();
+		ctx.add_arch(target_arch).unwrap();
+
+		//Deny these syscalls
+		for syscall in &[
+			"add_key",
+			"_sysctl",
+			"acct",
+			"add_key",
+			"adjtimex",
+			"chroot",
+			"clock_adjtime",
+			"create_module",
+			"delete_module",
+			"fanotify_init",
+			"finit_module",
+			"get_kernel_syms",
+			"get_mempolicy",
+			"init_module",
+			"io_cancel",
+			"io_destroy",
+			"io_getevents",
+			"io_setup",
+			"io_submit",
+			"ioperm",
+			"iopl",
+			"ioprio_set",
+			"kcmp",
+			"kexec_file_load",
+			"kexec_load",
+			"keyctl",
+			"lookup_dcookie",
+			"mbind",
+			"nfsservctl",
+			"migrate_pages",
+			"modify_ldt",
+			"mount",
+			"move_pages",
+			"name_to_handle_at",
```

Comment:
@bbx0 I have a question, can you please give a little bit of insight on why you included `name_to_handle_at` into the list of forbidden syscalls? From what I'm reading (`man 2 name_to_handle_at`), this seems fairly innocent, e.g. something that a legit build could do. Does this syscall have a history of being easy to exploit, or some other things about it?

Reply:
UPD: I see now that this is also now used in the upstream code that you've referenced. Same with `chroot`. I'll check there

Reply:
@vn971 Yes, I tried to reproduce the existing behavior of rua, which did not use the most recent version of upstream (but referenced commit). The current version of upstream seems also not deny `name_to_handle_at` or `chroot` anymore.